### ISSUE TITLE
[feedback] format mac does not work in swagger:strfmt

### ISSUE THREAD

**Issue by allencloud** [go-swagger/go-swagger#1348]
## Problem statement

I ran into an issue that I wish swagger could help me to validate MAC address in struct defined in swagger.yml, like:

```
      MacAddress:
        description: "MAC address of the container."
        type: "string"
        format: mac
```
Then the generated struct has a validation function to validate the mac format of this field.

However it fails.

Could we support this feature in go-swagger?

## Environment
swagger version: `version: 0.12.0`
go version: `go version go1.9.1 darwin/amd64` 
OS:

**Comment by fredbi** [comment]
Thanks. I've just added this in my current branch for spec validation part, that should merged in a couple days.
This will probably need some additional stuff in the generator part, though. Let us figure that out with @casualjim next week

**Comment by fredbi** [comment]
When analyzing the type validation, I could see there a quite a few other formats missing.
Tedious homework, but it should be done.

@casualjim : I don't know about the generator, but regarding the validate.typeValidator.schemaInfoType(), I suggest this kind of reverse lookup be deferred to strfmt : on the long term, it is just not possible to keep things in sync with all fmts, not to mention the possibility of custom format registries.

**Comment by fredbi** [comment]
The generation part should be available in go-swagger with PR#1368
The validation part should be available with PR go-openapi/validate#50

@allencloud with these live branches available (not yet merged), could you please confirm your issue will be solved?

**Comment by fredbi** [comment]
We need to update vendor before closing this
